src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNet.Server.Kestrel.Infrastructure
 {
     public struct MemoryPoolIterator2
     {
+        private const int _maxHeaderStackLength = 16384;
```

Comment:
For ref: 
nginx [default 8k]( http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers)
Apache [default 8190 bytes](http://httpd.apache.org/docs/2.4/mod/core.html#limitrequestfieldsize)
IIS default 16384 bytes